.github/workflows/release.yaml

```diff
@@ -35,175 +36,104 @@ jobs:
-      # - name: Set up QEMU
-      #   uses: docker/setup-qemu-action@v1
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+      - name: Export CURRENT_BRANCH env var
+        # can't use github.ref_name because that will be e.g. '2.6.9' on tag push, and we want to cache-from main branch on release
+        run: echo "CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD | sed 's/\//-/g')" >> $GITHUB_ENV
 
       - name: Generate image metadata
         id: metadata
         uses: docker/metadata-action@v4
         with:
-          images: prefecthq/prefect
+          images: ${{ github.repository }}
+          # list CURRENT_BRANCH first, to use as CACHE_IMAGE below
           tags: |
-            type=pep440,pattern={{version}},suffix=-python${{ matrix.python-version }}-conda
-            type=pep440,pattern={{major}}.{{minor}},suffix=-python${{ matrix.python-version }}-conda
-            type=pep440,pattern={{major}},suffix=-python${{ matrix.python-version }}-conda
-            type=sha,suffix=-python${{ matrix.python-version }}-conda
-            type=raw,value=2-latest-conda,enable=${{ matrix.python-version == '3.10'}}
+            type=raw,value=${{ env.CURRENT_BRANCH }},suffix=-python${{ matrix.python-version }}${{ matrix.flavor }}
+            type=sha,suffix=-python${{ matrix.python-version }}${{ matrix.flavor }}
+            type=raw,enable=${{ github.event_name == 'release' && matrix.python-version == '3.10' }},value=2-latest${{ matrix.flavor }}
+            type=pep440,enable=${{ github.event_name == 'release' }},pattern={{version}},suffix=-python${{ matrix.python-version }}${{ matrix.flavor }}
+            type=pep440,enable=${{ github.event_name == 'release' }},pattern={{major}}.{{minor}},suffix=-python${{ matrix.python-version }}${{ matrix.flavor }}
+            type=pep440,enable=${{ github.event_name == 'release' }},pattern={{major}},suffix=-python${{ matrix.python-version }}${{ matrix.flavor }}
           flavor: |
             latest=false
 
-      - name: Checkout
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
+      - name: Export CACHE_IMAGE env var
+        run: echo "CACHE_IMAGE=$(echo '${{ steps.metadata.outputs.json }}' | jq -r '.tags[0]')" >> $GITHUB_ENV
 
       - name: Build image
         uses: docker/build-push-action@v3
         with:
           context: .
+          platforms: linux/amd64,linux/arm64
           build-args: |
             PYTHON_VERSION=${{ matrix.python-version }}
-            BASE_IMAGE=prefect-conda
+            ${{ ( endsWith(matrix.flavor, 'conda') && 'BASE_IMAGE=prefect-conda' ) || '' }}
           tags: ${{ steps.metadata.outputs.tags }}
           labels: ${{ steps.metadata.outputs.labels }}
-          outputs: type=docker,dest=/tmp/image-conda.tar
-          cache-from: type=gha
-          # We do not cache Conda image layers because they very big and slow to upload
-          # cache-to: type=gha,mode=max
+          push: true
```

Comment:
like this, on push to main and on manual workflow_dispatch (can be triggered from Actions tab for any branch), it will push to docker.io:

- `prefecthq/prefect:main-python3.x`
- `prefecthq/prefect:main-python3.x-conda`
- `prefecthq/prefect:sha-424242-python3.x`
- `prefecthq/prefect:sha-424242-python3.x-conda`

as per the `metadata-action` step

adding a `-dev` prefix is possible in the `repository` key of that step based on `github.event_name` similar to the `BASE_IMAGE` conditional